alacritty_terminal/src/term/mod.rs

```diff
@@ -501,6 +501,57 @@ impl<T> Term<T> {
         }
     }
 
+    /// Go to next prompt farthest down in history.
+    pub fn goto_next_prompt(&mut self)
+    where
+        T: EventListener,
+    {
+        let point = if self.mode.contains(TermMode::VI) {
+            self.vi_mode_cursor.point
+        } else {
+            let display_offset = self.grid.display_offset() as i32;
+            // If we're outside of Vi mode use bottom most visible line.
+            Point::new(Line(self.screen_lines() as i32 - 1 - display_offset), Column(0))
```

Comment:
Why are you using the bottommost line? If I have a prompt visible at the top and another prompt in the middle of my screen, I want to jump to the prompt in the middle of my screen. I don't want to jump to the next offscreen prompt.

This is the same as search, the top of the screen is always to be considered the origin of the search.

This also shouldn't be inverted for `goto_previous`. Even there we should still make it based on the topmost line and not switch to the bottommost. We always want our search match aligned at the top of the viewport.